### PR TITLE
Add reject notifications

### PR DESCRIPTION
**Pull request by slifty**
This small PR is part of the work for Issue #341 -- it adds a notification to the act of rejecting an enrollment requests.

It does not involve any form of refactoring for those requests, which is something discussed in Issue #721, which is something that should be done in a future ticket if / when that type of cleanup becomes a priority.

### CODE REVIEW

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java [hunk 1]
```diff
@@ -160,6 +169,15 @@ public void rejectTicket(
             throw new PortalServiceException("Cannot change status because it is not in pending state.");
         }
 
+        // Look up the user who submitted the ticket so we can get their email address
+        Query query = getEm().createQuery("FROM CMSUser where username = :username");
+        query.setParameter("username", ticket.getSubmittedBy());
+        CMSUser submittingUser = (CMSUser) query.getSingleResult();
```
Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java]: A few small things about this block of code:
1. Would you mind extracting a method? `rejectTicket` is getting a bit long, and an introductory comment is a good clue that the code should instead live in its own, well-named method. :)
2. Please use the [`TypedQuery` version of `createQuery`](https://docs.oracle.com/javaee/7/api/javax/persistence/EntityManager.html#createQuery-java.lang.String-java.lang.Class-), so that you don't have to cast the result. I know we have a lot of bad examples of the `Query` version, but I've used the `TypedQuery` version in at least [`LookupServiceBean`](https://github.com/SolutionGuidance/psm/blob/bcab22c4ec8ccc5f7b2c7b2a9519a5c1b09d032f/psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/LookupServiceBean.java#L274-L282).
3. I believe these methods can all be chained, so that you never have to have a `Query` (or `TypedQuery`) variable.

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java [hunk 2]
```diff
@@ -160,6 +169,15 @@ public void rejectTicket(
...
+
+        Map<String, Object> vars = new HashMap<String, Object>();
```
Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java]: We're on Java 8, so you can use the [diamond operator](https://www.javaworld.com/article/2074080/core-java/jdk-7--the-diamond-operator.html): `... = new HashMap<>();`.

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java (hunk 1 above)
Comment (slifty) [reply psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java]: Good call -- I added a note reflecting some further frustration, which is that I'd suggest SQL never belongs outside of a model (in essence: these three lines will have bad odors to me no matter where in this Bean they live).

Doing right by the model pattern will be a somewhat more significant refactor I suspect (beyond the scope of this ticket, and worth doing comprehensively I suspect).

TL;DR -- absolutely happy to keep rejectTicket shorter, but let the record show that I feel bad about the block regardless!

Comment (jasonaowen) [reply psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java]: @slifty totally agree; the real fix would be to actually link the `Enrollment` and `CMSUser` entities. See also [my comment capturing this on #57](https://github.com/SolutionGuidance/psm/issues/57#issuecomment-382841067).

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java [hunk 3]
```diff
@@ -160,12 +169,37 @@ public void rejectTicket(
...
+     * Look up the user who submitted a ticket
+     *
+     * TODO: this is not the appropriate location for this functionality
+     * If / when a refactor to how we do our models is done, this should
+     * live inside of the Ticket model.
+     *
+     * @param  ticketId [description]
+     * @return          [description]
+     */
+    private CMSUser getUserByEnrollment(Enrollment ticket) {
```
Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java]: Can we replace "ticket" with "enrollment" or "enrollment application"? "Ticket" is a legacy term that seems to have been incompletely renamed before we started working on the PSM. (cc @cecilia-donnelly, since we've previously talked about renaming the `Enrollment` class to, iirc, `EnrollmentApplication` or similar)

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java [hunk 4]
```diff
@@ -160,12 +169,37 @@ public void rejectTicket(
...
 
+    /**
+     * Look up the user who submitted a ticket
+     *
+     * TODO: this is not the appropriate location for this functionality
+     * If / when a refactor to how we do our models is done, this should
+     * live inside of the Ticket model.
+     *
+     * @param  ticketId [description]
+     * @return          [description]
```
Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java]: No placeholder docs, please. Either fill out a useful description or skip the javadoc.

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java [hunk 5]
```diff
@@ -160,12 +169,37 @@ public void rejectTicket(
...
+     */
+    private CMSUser getUserByEnrollment(Enrollment ticket) {
+        return getEm().createQuery(
+            "FROM CMSUser where username = :username",
+            CMSUser.class
+        ).setParameter(
+            "username",
+            ticket.getSubmittedBy()
+        ).getSingleResult();
```
Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java]: :+1:

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java (hunk 4 above)
Comment (slifty) [reply psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java]: Complete accident, apologies for that and thank you for the catch.
